parts/kubernetesmastercustomdata.yml

```diff
@@ -249,6 +249,7 @@ write_files:
     sed -i "s|<kubernetesKubeDNSSpec>|{{WrapAsVariable "kubernetesKubeDNSSpec"}}|g; s|<kubernetesDNSMasqSpec>|{{WrapAsVariable "kubernetesDNSMasqSpec"}}|g; s|<kubernetesExecHealthzSpec>|{{WrapAsVariable "kubernetesExecHealthzSpec"}}|g" "/etc/kubernetes/addons/kube-dns-deployment.yaml"
     sed -i "s|<kubernetesHeapsterSpec>|{{WrapAsVariable "kubernetesHeapsterSpec"}}|g; s|<kubernetesAddonResizerSpec>|{{WrapAsVariable "kubernetesAddonResizerSpec"}}|g" "/etc/kubernetes/addons/kube-heapster-deployment.yaml"
     sed -i "s|<kubernetesDashboardSpec>|{{WrapAsVariable "kubernetesDashboardSpec"}}|g" "/etc/kubernetes/addons/kubernetes-dashboard-deployment.yaml"
+    sed -i "s|<kubeDnsServiceIP>|{{{kubeDnsServiceIP}}}|g" "/etc/kubernetes/addons/kube-dns-service.yaml"
```

Comment:
this line is wrong. It needs to look more like
>|{{WrapAsVariable "kubeDnsServiceIP"}}|

Then I think this should work. I'm guessing running the cloud-init caused issues at this line and caused it to not download the custom script